base/version.jl

```diff
@@ -107,7 +107,7 @@ end
 
 function tryparse(::Type{VersionNumber}, v::AbstractString)
     v == "∞" && return typemax(VersionNumber)
-    m = match(VERSION_REGEX, v)
+    m = match(VERSION_REGEX, String(v))
```

Comment:
`String(v)` doesn't necessarily infer to string, so we usually add an additional type assert:

```suggestion
    m = match(VERSION_REGEX, String(v)::String)
```